src/android/LocalNotification.java

```diff
@@ -313,7 +337,91 @@ private void scheduleWithPermission(){
                 fireEvent("add", toast);
             }
         }
-        success(callbackContext, true);
+
+        sendScheduleResult(callbackContext, PluginResult.Status.OK, warning);
+        warning = null;
+
+    }
+
+    /**
+     * Checks if there's any notification to be scheduled as exact.
+     *
+     * @return true if there's at least one exact notification. false otherwise.
+     */
+    private boolean hasAnyExactNotification() {
+        for (int i = 0; i < notificationArguments.length(); i++) {
+            JSONObject dict    = notificationArguments.optJSONObject(i);
+            Options options    = new Options(cordova.getActivity(), dict);
+            if(options.getIsExactNotification()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Checks if there's any notification that is mandatory to be scheduled as exact.
+     *
+     * @return true if there's at least one mandatory notification. false otherwise.
+     */
+    private boolean hasAnyMandatoryExactNotification() {
```

Comment:
Can't we join this with `hasAnyExactNotification`? Besides the method used in `if` statement, it's identical.

Reply:
We actually can! Consider it done.